### DISCUSSION TITLE
DAI 2024-2025 - Practical work 2 - Yoann Changanaqui, Camille Theubet

### DISCUSSION THREAD

**Discussion by CamilleTCH** [heig-vd-dai-course/heig-vd-dai-course#489] [Show and tell]
We want to create a tool that can create playlists that appeal to a small set of participants. One participant create a listening group on a remote server, the other ones connect to it. Once everyone is in the group, the admin tells the server to start, and then the server creates a common playlist (as text) based on the preferences of everyone. Each participant's preferences are stored in a simple text file (format yet to determine) locally on their computer. Clients and server will communicate with TCP to exchange each user's preferences.

- client side : edit self preferences, create a remote group (and therefore be its admin), or join/quit a group.

- server side : manage every user's connection to a group, create the collective playlist by asking every client it's local preferences list.

To start simple, we will only allow the user the set the preferences between a few musical styles, each musical style having a few defined songs attached to it (the choice of the songs are based on our personnal **objectives** preferences).

**Comment by ludelafo** [comment]
Looks good for us! Thank you!

**Comment by CamilleTCH** [comment]
Hello,

Link to our repository : [consuly](https://github.com/CamilleThHEIG/consuly)
Here is the commit hash for the evaluation : 2646359974dc7179ea48fd8c9d83edb1d7608201 [Link to hash](https://github.com/CamilleThHEIG/consuly/commit/2646359974dc7179ea48fd8c9d83edb1d7608201)
As a bonus : tag to the final commit  :https://github.com/CamilleThHEIG/consuly/releases/tag/rendu

See you tommorow :D
